keylime/da/record.py

```diff
@@ -0,0 +1,375 @@
+        """Encodes record"""
+
+        _encoded_record_object = record_object
+
+        if self.rcd_fmt == "pickle":
+            _encoded_record_object = pickle.dumps(record_object)
+
+        if self.rcd_fmt == "json":
+            _encoded_record_object = json.dumps(record_object, indent=4)
+            if self.rcd_enc == "base64":
+                _encoded_record_object = str.encode(_encoded_record_object)
+
+        if self.rcd_enc == "base64":
+            _encoded_record_object = base64.b64encode(_encoded_record_object)
+
+        return _encoded_record_object
+
+    def _record_decode(self, record_object):
+        """Decodes record"""
+
+        _decoded_record_object = record_object
+
+        if self.rcd_enc == "base64":
+            _decoded_record_object = base64.b64decode(_decoded_record_object)
+
+        if self.rcd_fmt == "pickle":
+            _decoded_record_object = pickle.loads(_decoded_record_object)
+
+        if self.rcd_fmt == "json":
+            if self.rcd_enc == "base64":
+                _decoded_record_object = _decoded_record_object.decode("utf-8")
+            _decoded_record_object = json.loads(_decoded_record_object)
+
+        return _decoded_record_object
+
+    def _record_sanitize(self, record_object):
+        """Removes complex python objects from record"""
+        for _key in self.rmv_a:
+            if _key in record_object["agent"]:
+                record_object["agent"][_key] = None
+
+    def _record_assemble(self, record_object, agent_data, attestation_data, ima_policy_data):
+        """Assemble record to be created on persistent datastore"""
+        if "agent" not in record_object:
+            record_object["agent"] = {}
+            record_object["agent"].update(agent_data)
+
+        self._record_sanitize(record_object)
+
+        if attestation_data:
+            record_object["json_response"] = attestation_data
+
+        if ima_policy_data:
+            record_object["ima_policy"] = ima_policy_data.ima_policy
+
+    def _record_assemble_for_signing(self, record_to_sign, agent_data, attestation_data, signed_attributes):
+        """Builds a dictionary of attributes to dumped into a JSON file to be signed"""
+        if not signed_attributes:
+            return None
+
+        if signed_attributes == "all":
+            record_to_sign["agent"] = {}
+            record_to_sign["agent"].update(agent_data)
+
+            if attestation_data:
+                record_to_sign["json_response"] = {}
+                record_to_sign["json_response"].update(attestation_data)
+        else:
+            try:
+                signed_attributes = signed_attributes.split(",")
+            except Exception:
+                pass
+
+            record_to_sign["agent"] = {}
+            for _attr in signed_attributes:
+                if _attr in agent_data:
+                    record_to_sign["agent"][_attr] = agent_data[_attr]
+
+        self._record_sanitize(record_to_sign)
+        return None
```

Comment:
?

Reply:
This still needs to be addressed. The return type annotation is bool but returns only None.
Also I would expect it to return a copy of the record with the added values and not mutate it, but that is more a style choice.

Reply:
done